src/pages/index.js

```diff
@@ -1,65 +1,136 @@
-const addCardTitleInput = document.querySelector("#card-title-textbox");
-const addCardUrlInput = document.querySelector("#card-subtitle-textbox");
-
-const config = {
-  formSelector: ".modal__form",
-  inputSelector: ".modal__text-input",
-  submitButtonSelector: ".modal__button",
-  inactiveButtonClass: "modal__button-inactive",
-  inputErrorClass: "modal__invalid",
-  errorClass: "modal__invalid_active",
-  inputInvalidClass: "modal__text_invalid",
-};
+});
+
+// Function to handle like button click
+function handleLikeButtonClick(card) {
+  if (card.isLiked()) {
+    api
+      .dislikeCard(card._id)
+      .then((res) => {
+        card.setIsLiked(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(card._id)
+      .then((res) => {
+        card.setIsLiked(res.isLiked);
+      })
+      .catch(console.error);
+  }
+}
+
+// Function to handle delete button click
+function handleDeleteButtonClick(card) {
+  deleteModalPopup.setConfirm(() => {
+    deleteModalPopup.renderLoading(true);
+    api
+      .deleteCard(card._id)
+      .then((res) => {
+        card.handleDeleteButton(res);
+        deleteModalPopup.close();
+      })
+      .catch((error) => {
+        console.error("Error deleting card:", error);
+      })
+      .finally(() => {
+        deleteModalPopup.renderLoading(false);
+      });
+  });
+  deleteModalPopup.open();
+}
+
+// Creating an instance of PopupWithForm for delete modal
+const deleteModalPopup = new PopupDeleteConfirm(
+  "#delete-modal-popup",
+  handleDeleteButtonClick
+);
+
+deleteModalPopup.setEventListeners();
+
+// Function to handle avatar form submission
+function handleAvatarFormSubmit(data) {
+  avatarModalPopup.renderLoading(true);
+
+  api
+    .updateAvatar(data)
+    .then((res) => {
+      console.log("Avatar updated successfully");
+      userInfo.setAvatar(res.avatar);
+      avatarModalPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error updating avatar:", error);
+    })
+    .finally(() => {
+      avatarModalPopup.renderLoading(false);
+    });
+}
+
+// Event listener for avatar icon click to open avatar modal
+avatarIcon.addEventListener("click", handleAvatarClick);
 
+const avatarModalPopup = new PopupWithForm(
+  "#modal-avatar-popup",
+  handleAvatarFormSubmit
+);
+
+function handleAvatarClick() {
+  avatarModalPopup.open();
+}
+
+avatarModalPopup.setEventListeners();
+
+// Function to handle card form submission
 function handleCardFormSubmit() {
-  const name = addCardTitleInput.value;
-  const link = addCardUrlInput.value;
-  renderCard({ name, link });
-  cardFormValidator.disableButton();
-  addCardPopup.close();
+  addCardPopup.renderLoading(true);
+
+  api
+    .addCard({ name: addCardTitleInput.value, link: addCardUrlInput.value })
+    .then((cardData) => {
+      console.log("Card added successfully:");
+      renderCard(cardData);
```

Comment:
Needs correction: please make sure that the new card has an id. Now the new cards can't be liked or removed